gravitee-apim-console-webui/src/management/api/creation/v4/models/ApiCreationStepperService.ts

```diff
@@ -48,35 +50,45 @@ export class ApiCreationStepperService {
    */
   public steps$ = new ReplaySubject<ApiCreationStep[]>(1);
 
-  constructor(payload: ApiCreationStepPayload[]) {
-    this.steps = payload.map((stepPayload) => ({
-      payload: {},
+  constructor(creationStepPayload: ApiCreationStepPayload[], initialPayload?: ApiCreationPayload) {
+    this.initialPayload = initialPayload ?? {};
+    this.steps = creationStepPayload.map((stepPayload) => ({
       ...stepPayload,
+      patchPayload: (p) => p,
       type: 'primary',
     }));
     this.steps$.next(this.steps);
   }
 
-  goToStep(index: number, payload?: ApiCreationPayload) {
+  /**
+   * This function compile the data for a specific step of the API creation process by using the data from previous steps
+   *
+   * @param {ApiCreationStep} step - the current step of the API creation process
+   * @returns {ApiCreationPayload} - the data for the current step
+   */
+  public compileStepPayload(step: ApiCreationStep): ApiCreationPayload {
+    // Get all the previous steps of the current step
+    const previousSteps = this.steps.slice(0, this.steps.indexOf(step));
+    // Compute the data for the current step by using reduce method on the previous steps
+    // The reduce method is used to combine all the data from the previous steps using the patchPayload method
+    // The initial payload is used as the starting value for the reduce method
+    return step.patchPayload(previousSteps.reduce((payload, previousStep) => previousStep.patchPayload(payload), this.initialPayload));
```

Comment:
To simplify this line, create a list with previous steps and the 'step' parameter at the end, then reduce.